tests/test_parser_steps.py

```diff
@@ -0,0 +1,600 @@
+                '    ',
+                '    When I call get_step on an empty list',
+            ],
+        )
+
+    def test_getting_raw_step(self):
+        """
+            Check we can get a step with no surrounding data.
+        """
+        # Given I have Romaine core's parser
+        parser = common.get_romaine_parser()
+
+        # When I call get_step on a list containing
+        # """[
+        #     'When I call get_step on an empty list'
+        # ] """
+        input_data = [
+            'When I call get_step on an empty list',
+        ]
+        result = parser.step.get_step(input_data)
+
+        # Then the result is a step with type=When and
+        # text='I call get_step on an empty list' and nothing remaining
+        self.assertEqual(
+            result,
+            {
+                'step': {
+                    'leading_comments_and_space': [],
+                    'type': 'When',
+                    'text': 'I call get_step on an empty list',
+                    'multiline_arg': None,
+                    'trailing_whitespace': [],
+                },
+                'remaining': [],
+                'raw_input': input_data,
+            },
+        )
+
+    def test_getting_commented_step(self):
+        """
+            Check we can get a step with a leading comment.
+        """
+        # Given I have Romaine core's parser
+        parser = common.get_romaine_parser()
+
+        # When I call get_step on a list containing
+        # """[
+        #     '    # Check for empty steps',
+        #     '    ',
+        #     'When I call get_step on an empty list'
+        # ] """
+        input_data = [
+            '    # Check for empty steps',
+            '    ',
+            '    When I call get_step on an empty list',
+        ]
+        result = parser.step.get_step(input_data)
+
+        # Then the result is a step with type=When and
+        # text='I call get_step on an empty list',
+        # leading_comments_and_space=['    # Check for empty steps',
+        # '    '], and nothing remaining
+        self.assertEqual(
+            result,
+            {
+                'step': {
+                    'leading_comments_and_space': [
+                        '    # Check for empty steps',
+                        '    ',
+                    ],
+                    'type': 'When',
+                    'text': 'I call get_step on an empty list',
+                    'multiline_arg': None,
+                    'trailing_whitespace': [],
+                },
+                'remaining': [],
+                'raw_input': input_data,
+            },
+        )
+
+    def test_get_step_empty(self):
+        """
+            Confirm that we get nothing when we don't start with a step.
+        """
+        # Given I have Romaine core's parser
+        parser = common.get_romaine_parser()
+
+        # When I call get_step on a list containing:
+        # """[
+        #     'This is not a real step.',
```

Comment:
How so? This is only called where it should get steps.

Probably best if you can come up with a test that shows when it's a good idea for this not to happen so that we can code against it?